common.c

```diff
@@ -844,6 +932,91 @@ static void grabPointer(X11ConnData* data, Window window)
 	data->notes[serial] = Note_X_GrabPointer;
 }
 
+static void X11Handshake(void* buf, size_t length)
+{
+	size_t usedbytes = sz_xConnSetup;
+	if (length < usedbytes)
+	{
+		log_debug2("malformed handshake: Too short\n");
+		return;
+	}
+	xConnSetup* c = (xConnSetup *)buf;
+	buf += sz_xConnSetup;
+	log_debug2(" xConnSetup vendor='%.*s' numRoots=%d numFormats=%d\n",
+			   c->nbytesVendor, (char*) buf, c->numRoots, c->numFormats);
+
+	// fakeScreenResolution
+	if (config.fakeScreenW <= 0
+	&& config.fakeScreenH <= 0
+	&& config.fakeScreenDimW <= 0
+	&& config.fakeScreenDimH <= 0)
```

Comment:
Nit, these variables are unsigned so they can never be below zero

Reply:
Completely forgot that they were unsigned integers. Will fix that.